diff -Naur a/wscript_configure_embedded_heimdal b/wscript_configure_embedded_heimdal
--- a/wscript_configure_embedded_heimdal	2022-10-02 22:34:22.306761500 +0700
+++ b/wscript_configure_embedded_heimdal	2022-10-02 22:35:34.915712800 +0700
@@ -6,3 +6,14 @@
 
 conf.define('USING_EMBEDDED_HEIMDAL', 1)
 conf.RECURSE('source4/heimdal_build')
+
+def check_system_heimdal_binary(name):
+    if conf.LIB_MAY_BE_BUNDLED(name):
+        return False
+    if not conf.find_program(name, var=name.upper()):
+        return False
+    conf.define('USING_SYSTEM_%s' % name.upper(), 1)
+    return True
+
+check_system_heimdal_binary("compile_et")
+check_system_heimdal_binary("asn1_compile")
diff -Naur a/wscript_configure_system_heimdal b/wscript_configure_system_heimdal
--- a/wscript_configure_system_heimdal	2022-10-02 22:34:01.547740400 +0700
+++ b/wscript_configure_system_heimdal	2022-10-02 22:36:02.443038200 +0700
@@ -37,14 +37,6 @@
     conf.define('USING_SYSTEM_%s' % name.upper(), 1)
     return True
 
-def check_system_heimdal_binary(name):
-    if conf.LIB_MAY_BE_BUNDLED(name):
-        return False
-    if not conf.find_program(name, var=name.upper()):
-        return False
-    conf.define('USING_SYSTEM_%s' % name.upper(), 1)
-    return True
-
 check_system_heimdal_lib("com_err", "com_right_r com_err", "com_err.h")
 
 if check_system_heimdal_lib("roken", "rk_socket_set_reuseaddr", "roken.h"):
@@ -96,9 +88,6 @@
 #if conf.CHECK_BUNDLED_SYSTEM('tommath', checkfunctions='mp_init', headers='tommath.h'):
 #    conf.define('USING_SYSTEM_TOMMATH', 1)
 
-check_system_heimdal_binary("compile_et")
-check_system_heimdal_binary("asn1_compile")
-
 conf.env.KRB5_VENDOR = 'heimdal'
 conf.define('USING_SYSTEM_KRB5', 1)
 conf.define('USING_SYSTEM_HEIMDAL', 1)
